[2025-07-06T15:21:13.305+0100] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-07-06T15:21:13.321+0100] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: interviewgraphs_process_youtube_data.map_category_names manual__2025-07-06T14:15:01.383897+00:00 [queued]>
[2025-07-06T15:21:13.327+0100] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: interviewgraphs_process_youtube_data.map_category_names manual__2025-07-06T14:15:01.383897+00:00 [queued]>
[2025-07-06T15:21:13.328+0100] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2025-07-06T15:21:13.337+0100] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): map_category_names> on 2025-07-06 14:15:01.383897+00:00
[2025-07-06T15:21:13.340+0100] {standard_task_runner.py:64} INFO - Started process 6573 to run task
[2025-07-06T15:21:13.342+0100] {standard_task_runner.py:90} INFO - Running: ['airflow', 'tasks', 'run', 'interviewgraphs_process_youtube_data', 'map_category_names', 'manual__2025-07-06T14:15:01.383897+00:00', '--job-id', '86', '--raw', '--subdir', 'DAGS_FOLDER/interview_graphs.py', '--cfg-path', '/tmp/tmpi_wkru6w']
[2025-07-06T15:21:13.342+0100] {standard_task_runner.py:91} INFO - Job 86: Subtask map_category_names
[2025-07-06T15:21:13.385+0100] {task_command.py:426} INFO - Running <TaskInstance: interviewgraphs_process_youtube_data.map_category_names manual__2025-07-06T14:15:01.383897+00:00 [running]> on host donald-VirtualBox
[2025-07-06T15:21:13.446+0100] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='interviewgraphs_process_youtube_data' AIRFLOW_CTX_TASK_ID='map_category_names' AIRFLOW_CTX_EXECUTION_DATE='2025-07-06T14:15:01.383897+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-07-06T14:15:01.383897+00:00'
[2025-07-06T15:21:13.447+0100] {taskinstance.py:430} INFO - ::endgroup::
[2025-07-06T15:21:13.602+0100] {logging_mixin.py:188} WARNING - /home/donald/airflow/dags/interview_graphs.py:162 FutureWarning: Passing literal json to 'read_json' is deprecated and will be removed in a future version. To read from a literal string, wrap it in a 'StringIO' object.
[2025-07-06T15:21:13.751+0100] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-07-06T15:21:13.752+0100] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/donald/Documents/Tutorials/ResponseMediaTest/.venv/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/donald/Documents/Tutorials/ResponseMediaTest/.venv/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/donald/Documents/Tutorials/ResponseMediaTest/.venv/lib/python3.11/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/donald/Documents/Tutorials/ResponseMediaTest/.venv/lib/python3.11/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/donald/Documents/Tutorials/ResponseMediaTest/.venv/lib/python3.11/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/donald/airflow/dags/interview_graphs.py", line 170, in map_category_names
    df.to_csv(save_path, index=False)
  File "/home/donald/Documents/Tutorials/ResponseMediaTest/.venv/lib/python3.11/site-packages/pandas/util/_decorators.py", line 333, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/donald/Documents/Tutorials/ResponseMediaTest/.venv/lib/python3.11/site-packages/pandas/core/generic.py", line 3986, in to_csv
    return DataFrameRenderer(formatter).to_csv(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/donald/Documents/Tutorials/ResponseMediaTest/.venv/lib/python3.11/site-packages/pandas/io/formats/format.py", line 1014, in to_csv
    csv_formatter.save()
  File "/home/donald/Documents/Tutorials/ResponseMediaTest/.venv/lib/python3.11/site-packages/pandas/io/formats/csvs.py", line 251, in save
    with get_handle(
         ^^^^^^^^^^^
  File "/home/donald/Documents/Tutorials/ResponseMediaTest/.venv/lib/python3.11/site-packages/pandas/io/common.py", line 873, in get_handle
    handle = open(
             ^^^^^
IsADirectoryError: [Errno 21] Is a directory: '/home/donald/Documents/Tutorials/ResponseMediaTest'
[2025-07-06T15:21:13.775+0100] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=interviewgraphs_process_youtube_data, task_id=map_category_names, run_id=manual__2025-07-06T14:15:01.383897+00:00, execution_date=20250706T141501, start_date=20250706T142113, end_date=20250706T142113
[2025-07-06T15:21:13.787+0100] {standard_task_runner.py:110} ERROR - Failed to execute job 86 for task map_category_names ([Errno 21] Is a directory: '/home/donald/Documents/Tutorials/ResponseMediaTest'; 6573)
[2025-07-06T15:21:13.833+0100] {local_task_job_runner.py:243} INFO - Task exited with return code 1
[2025-07-06T15:21:13.848+0100] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-07-06T15:21:13.850+0100] {local_task_job_runner.py:222} INFO - ::endgroup::
